commons-impl/src/main/java/com/connexta/transformation/commons/inmemory/InMemoryTransformationManager.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.commons.inmemory;
+
+import com.connexta.transformation.commons.api.TransformationManager;
+import com.connexta.transformation.commons.api.exceptions.TransformationException;
+import com.connexta.transformation.commons.api.exceptions.TransformationNotFoundException;
+import com.connexta.transformation.commons.api.status.MetadataTransformation;
+import com.connexta.transformation.commons.api.status.Transformation;
+import com.connexta.transformation.commons.inmemory.status.InMemoryTransformation;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InMemoryTransformationManager implements TransformationManager {
+  private final Map<String, Transformation> store = new HashMap<>();
+
+  @Override
+  public Transformation createTransform(
+      URI currentLocation, URI finalLocation, URI metadataLocation) throws TransformationException {
+    Transformation transformation =
+        new InMemoryTransformation(currentLocation, finalLocation, metadataLocation);
+    store.put(transformation.getTransformId(), transformation);
+    return transformation;
+  }
+
+  @Override
+  public Transformation get(String transformId) throws TransformationException {
+    Transformation transformation = store.get(transformId);
+    if (transformation == null) {
+      throw new TransformationNotFoundException(
+          "Transformation [" + transformId + "] cannot be found");
+    } else {
+      return transformation;
+    }
+  }
+
+  @Override
+  public MetadataTransformation get(String transformId, String type)
+      throws TransformationException {
+    return get(transformId)
+        .metadatas()
+        .filter(m -> m.getMetadataType().equals(type))
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new TransformationNotFoundException(
+                    "No [" + type + "] metadata found for transformation [" + transformId + "]"));
+  }
+
+  @Override
+  public void delete(String transformId) throws TransformationException {
+    Transformation transformation = store.remove(transformId);
+    if (transformation == null) {
```

Comment:
NOTE: Technically, a consumer could still modify and use references to a `Transformation` or `MetadataTransformation` object. Those references just wouldn't be accessible by other consumers. I'm not sure how we could clear those. Should we have an internal "tombstone" on all the objects, and any operations throw an NotFound exception if the tombstone is set?

Reply:
I may be missing something but I think whatever processes are using those objects could finish whatever they are doing. The result they produce just wouldn't be used anywhere.

Reply:
What is typically done is to marked the `Transformation` object as deleted such that operations done on it could detect that and throw error. Same for the metadata objects inside that transaction